### PR TITLE
Make `Application.ForeignMessageNames()` and `RichApplication.ForeignMessageTypes()` global.

### PR DESCRIPTION
**Pull request by jmalloc**
This allows us to compute the "foreign" messages from any implementation of `Application` / `RichApplication`.

### CODE REVIEW

File: application.go [hunk 1]
```diff
@@ -104,6 +77,92 @@ func IsApplicationEqual(a, b Application) bool {
...
+//  - events that are consumed by the application, but produced elsewhere
+func ForeignMessageNames(app Application) EntityMessageNames {
+	m := app.MessageNames()
+	f := EntityMessageNames{
+		Roles:    message.NameRoles{},
+		Produced: message.NameRoles{},
+		Consumed: message.NameRoles{},
+	}
+
+	for mt, r := range m.Produced {
```
Comment (koden-km) [inline application.go]: This is a minor thing, but is `mt` a good name for that var. These are "message names" right?
I see the `mt` "message type" version in the other function below.

File: application.go [hunk 2]
```diff
@@ -104,6 +77,92 @@ func IsApplicationEqual(a, b Application) bool {
...
+			f.Produced.Add(mt, r)
+		}
+	}
+
+	for mt, r := range m.Consumed {
+		if m.Produced.Has(mt) {
+			continue
+		}
+
+		// Any message type is considered foreign if it needs to be obtained from
```
Comment (koden-km) [inline application.go]: Are these actually "message type"'s as the comment mentions?

File: application.go (hunk 1 above)
Comment (jmalloc) [reply application.go]: Yep, fair call, this is a typical copy/paste error >_<

Comment (jmalloc) [reply application.go]: I've fixed these, improved the comments, and collapsed some verbose logic down a little. Please re-review.

File: application.go (hunk 2 above)
Comment (jmalloc) [reply application.go]: My terminology is possibly a bit lax, but yes -- the message "name" refers to the name of the type. I've reworded it anyway.

**Pull request merged**